extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerProjectImporter.java

```diff
@@ -0,0 +1,103 @@
+package com.microsoft.gradle.bs.importer;
+
+import static org.eclipse.jdt.ls.core.internal.handlers.MapFlattener.getValue;
+
+import java.nio.file.Path;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.jdt.ls.core.internal.AbstractProjectImporter;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.jdt.ls.core.internal.managers.BasicFileDetector;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+
+import ch.epfl.scala.bsp4j.BuildClientCapabilities;
+import ch.epfl.scala.bsp4j.BuildServer;
+import ch.epfl.scala.bsp4j.InitializeBuildParams;
+import ch.epfl.scala.bsp4j.InitializeBuildResult;
+
+public class GradleBuildServerProjectImporter extends AbstractProjectImporter {
+
+    private static final String CLIENT_NAME = "jdtls";
+
+    private static final String CLIENT_VERSION = "0.1.0";
+
+    private static final String BSP_VERSION = "2.1.0-M4";
+
+    public static final String BUILD_GRADLE_DESCRIPTOR = "build.gradle";
+    public static final String BUILD_GRADLE_KTS_DESCRIPTOR = "build.gradle.kts";
+    public static final String SETTINGS_GRADLE_DESCRIPTOR = "settings.gradle";
+    public static final String SETTINGS_GRADLE_KTS_DESCRIPTOR = "settings.gradle.kts";
+
+    private static final String JAVA_BUILD_SERVER_GRADLE_ENABLED = "java.gradle.buildServer.enabled";
+
+    @Override
+    public boolean applies(IProgressMonitor monitor) throws OperationCanceledException, CoreException {
+        if (rootFolder == null) {
```

Comment:
```suggestion
    if (StringUtils.isEmpty(rootFolder))...

```

Reply:
rootFolder is a `File` instance inherited from its parent.